### PR TITLE
Consumer / Publisher non-blocking init

### PR DESCRIPTION
**Pull request by nsweeting**
This sets up a Consumer / Publisher to be non-blocking during the init callback.

Related to issue #64.

There is a chance of creating a race condition with the current setup of doing `send(self(), :init)`. The new `GenServer.handle_continue/2` would be ideal for this situation. But that would require bumping the minimum elixir version to 1.7.

### CODE REVIEW

**Comment by mkorszun** [comment]
@nsweeting Great. Will try to review it as quick as possible.

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,6 @@ erl_crash.dump
 
 # Also ignore archive artifacts (built via "mix archive.build").
 *.ez
+
+# Ignore Elixir Language Server files
+/.elixir_ls
```
Comment (mkorszun) [inline .gitignore]: Lets add new line at the end of file

Comment (nsweeting) [reply .gitignore]: Done :+1:

**Comment by mkorszun** [comment]
For the future: https://github.com/meltwater/gen_rmq/issues/72

**Comment by mkorszun** [comment]
@nsweeting thanks for the PR - great job 👍

**Pull request merged**